atd-vzv/Dockerfile

```diff
@@ -0,0 +1,10 @@
+FROM node:16-bullseye
+RUN apt-get update
+RUN apt-get install -y vim aptitude magic-wormhole
+WORKDIR /root/
+RUN curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.1/install.sh | bash
+RUN npm install -g npm-check-updates
+WORKDIR /root/atd-vzv
+#COPY . /root/atd-vzv
+#RUN npm install
```

Comment:
I toyed with the idea of copying the state of the app at image build time into the docker image, so it could stand-alone. With the app on there, you can still bind-mount over it, and get the behavior we're used to, but it makes build times muuuch longer, and it also introduces a host of confusions when things are not working just right. 

I'd be curious if there's interest in this. Maybe something to experiment more with if we ever want to host our own web server on ECS instead of netlify, which seems very unlikely.

Reply:
I'm a bit torn about local dev of VZE and VZV with the extra Docker layer. I could be missing something so eager to hear other opinions, but my thoughts are:
- extra complexity
- you have to attach to the container to see status of the dev server, compile warnings, etc. which I depend on a lot
- npm start by itself is pretty straightforward and is the way we consistently start up other Create React App bootstrapped apps

Reply:
## TLDR

* Mike is right, and there isn't an obviously compelling reason to dockerize the apps.
* They still work the old way in this configuration, so I want to give it a whirl -- using docker for the VZV/VZE is optional.

## The deets

@mddilley's observations here are each entirely correct and well considered. I don't have a problem to point to that is explicitly solved by pushing the VZV and VZE down into a docker container, except maybe just the intent of "dockerize all the things" which is far from a cogent and valid reason. Dockerizing the app isn't really solving a problem we're bumping into and is more like a design philosophy, which is totally hand-wavy. 👋🌊

But, that said, I would like to propose that it stay in, at least for the time being, because the way its setup doesn't preclude the use of running the environments on the bare metal outside of docker. The code is bind-mounted into the containers, so you can both change into the root directory of either app and `npm start` them up, or you can use a docker container to run them, if that's your preference. I can see a world where a developer running on windows needs to get these spun up, and suddenly that docker environment is looking pretty enticing. (I'm reaching here, but I think it's a valid argument.)

I do see the concern that we'll make some change to the apps that would break the docker builds, but I don't feel like this will be a common occurrence. If it does happen, it probably means we've done something "interesting," and fixing the issue will likely be a revealing, learning experience. I propose that keeping the docker images working can be my responsibility and will not block a PR that otherwise would be accepted from being merged. This will help me learn if this is worth it or not, and we can go from there.

Reply:
I agree with Mike that I can see myself missing the compile warnings and status of the dev server, I found myself waiting to see the feedback one usually gets from `npm start` so its really good for you to point out Frank that we can still use the npm start to get them going. 

I need to dig into this more, but if we are running the backend locally, changing into the root directory and then using`npm start` still points to the staging environment right?

Reply:
That's a great question Chia. 🙏🏻 I believe this would be controlled by the `.env` file in place, and I think it would be helpful for me to add a `.env.local` similarly to how we have `.env.staging` and `.env.production`, for the VZE. I'll do that. Thanks!

Reply:
Y'all are on it! I didn't even think of what it would take to get this hooked up to a separate local server for the frontend. Agreed that optional is the way to go! 🙏 🚀